lib/cocoapods/target/pod_target.rb

```diff
@@ -430,13 +430,14 @@ def xcframeworks
     end
 
     # @return [Hash{String=>Array<String>}] The resource and resource bundle paths this target depends upon keyed by
-    #         spec name. Resources for app specs and test specs are directly added to “Copy Bundle Resources” phase
-    #         from the generated targets for frameworks, but not libraries. Therefore they are not part of the resource paths.
+    #         spec name. Resources (not resource bundles) for app specs are directly added to “Copy Bundle Resources”
+    #         phase which is not true for test specs or libraries, therefore they are not part of the resource paths for
+    #         app specs.
     #
     def resource_paths
       @resource_paths ||= begin
         file_accessors.each_with_object({}) do |file_accessor, hash|
-          resource_paths = if file_accessor.spec.non_library_specification? && build_as_framework?
+          resource_paths = if file_accessor.spec.app_specification?
```

Comment:
why is this only app specs, and not also test specs?

Reply:
it is wrong. It turns out _we_ do add those directly to the target if we are using frameworks. I will change.